api4/config.go

```diff
@@ -227,3 +241,47 @@ func patchConfig(c *Context, w http.ResponseWriter, r *http.Request) {
 	w.Header().Set("Cache-Control", "no-cache, no-store, must-revalidate")
 	w.Write([]byte(c.App.GetSanitizedConfig().ToJson()))
 }
+
+func makeFilterConfigByPermission(accessType filterType) func(c *Context, structField reflect.StructField, parentStructFieldName string) bool {
+	return func(c *Context, structField reflect.StructField, parentStructFieldName string) bool {
+		permissionMap := map[string]*model.Permission{}
+		for _, p := range model.AllPermissions {
+			permissionMap[p.Id] = p
+		}
```

Comment:
might be worth adding tests for this function?

Reply:
The closures it creates (bot read and write) are both tested.